mmf/models/mmf_transformer.py

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+from copy import deepcopy
+from typing import Any, Dict, Type
+
+import torch
+from mmf.common.registry import registry
+from mmf.common.typings import DictConfig
+from mmf.models.transformers.base import (
+    BaseTransformer,
+    BaseTransformerConfigType,
+    BaseTransformerInput,
+)
+from mmf.modules.encoders import MultiModalEncoderBase
+from torch import Tensor, nn
+from transformers.modeling_bert import BertPooler, BertPredictionHeadTransform
+
+
+class ImageEncoder(MultiModalEncoderBase):
+    """Extends the MultiModalEncoderBase class which builds the encoder based on
+    the config parameters. We can set the type of image encoder(resnet50, resnet152,
+    resnext50 etc) and other parameters like num of features, type of pooling etc.
+    """
+
+    def __init__(self, config: DictConfig):
+        super().__init__(config)
+
+    def build(self):
+        self.encoder = self._build_modal_encoder(self.config.image_encoder)
+
+    def forward(self, x: Tensor) -> Tensor:
+        return self.encoder(x)
+
+
+class MMFTransformerEmbeddings(nn.Module):
+    """Embedding class that can take any number of image or text modalities, each can
+    have their input id, position id and segment id. We generate embeddings of
+    dimension config.hidden_size for each and then first add the three embeddings
+    for each modality to have a modality specific embedding. We then concat the
+    modality specific embeddings to have a joint embedding.
+    """
+
+    def __init__(
+        self,
+        model_config: Dict[str, Any],
+        transformer_config: Dict[str, Any],
```

Comment:
Is this hugging face config?

Reply:
Yes, base transformer config. I would keep the name as is now.